include/boost/numeric/ublas/tensor/static_strides.hpp

```diff
@@ -199,40 +199,39 @@ struct basic_static_strides<basic_static_extents<T,Extents...>, Layout>
    * @returns the element at given pos
    */
   [[nodiscard]] inline 
-  constexpr auto at(size_type k) const 
+  constexpr const_reference at(size_type k) const 
   {
     return m_data.at(k);
   }
 
   [[nodiscard]] inline 
-  constexpr auto operator[](size_type k) const { return m_data[k]; }
+  constexpr const_reference operator[](size_type k) const noexcept { return m_data[k]; }
 
   //@returns the rank of basic_static_extents
   [[nodiscard]] inline 
-  constexpr auto size() const noexcept { return _size; }
+  constexpr size_type size() const noexcept { return static_cast<size_type>(_size); }
 
-    value_type back () const{
-        return m_data.back();
-    }
+  [[nodiscard]] inline
+  constexpr const_reference back () const noexcept{
+      return m_data.back();
+  }
 
   // default constructor
-  constexpr basic_static_strides(){
-            if constexpr( _size == 0 ){
-          return;
-        }else{
-          static_assert( static_traits::is_valid_v<extents_type>, "Error in boost::numeric::ublas::basic_static_strides() : shape is not valid."); 	
-
-          if constexpr( static_traits::is_vector_v<extents_type> || static_traits::is_scalar_v<extents_type> ){
-            return;
-          }else{
-            static_assert(_size >= 2, "Error in boost::numeric::ublas::basic_static_strides() : size of strides must be greater or equal 2.");
-          }
-        }
-
+  constexpr basic_static_strides() noexcept{
+    if constexpr( _size == 0 ){
+      return;
+    }else{
+      static_assert( static_traits::is_valid_v<extents_type>, "Error in boost::numeric::ublas::basic_static_strides() : shape is not valid."); 	
```

Comment:
but the problem is if the size is 0 it will assert. That is the reason I have put it inside the constexpr if.

Reply:
> but the problem is if the size is 0 it will assert. That is the reason I have put it inside the constexpr if.

the `static_assert` respects that by 
`_size == 0 || static_traits::is_valid_v<extents_type>`

Reply:
ok.